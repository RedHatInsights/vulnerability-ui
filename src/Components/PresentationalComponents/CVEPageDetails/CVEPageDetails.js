import { Grid, GridItem } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import { CVEPageContext } from '../../SmartComponents/CVEDetailsPage/CVEDetailsPage';
import CVEPageDetailsDescription from '../CVEPageDetailsDescription/CVEPageDetailsDescription';
import CVEPageDetailsSeverity from '../CVEPageDetailsSeverity/CVEPageDetailsSeverity';
import WithLoader from '../WithLoader/WithLoader';
import CSAwRuleBox from '../CSAwRuleBox/CSAwRuleBox';

const CVEPageDetails = ({ data, changeExposedSystemsParameters }) => {
    return (
        <CVEPageContext.Consumer>
            {context => (
                <Grid hasGutter>
                    <GridItem md={8} sm={12}>
                        <WithLoader loading={context.isLoading} variant="spinner">
                            <CVEPageDetailsDescription cveAttributes={data.data} />
                        </WithLoader>
                    </GridItem>

                    <GridItem md={4} sm={12} className="pf-u-mt-sm pf-u-ml-sm">
                        <CVEPageDetailsSeverity cveAttributes={data.data} />
                    </GridItem>
                    <CSAwRuleBox
                        changeExposedSystemsParameters = {changeExposedSystemsParameters}
                        synopsis={data.data.synopsis}
                        rules={data.data.rules}
                    />
                </Grid>
            )}
        </CVEPageContext.Consumer>
    );
};

CVEPageDetails.propTypes = {
    data: propTypes.object,
    changeExposedSystemsParameters: propTypes.func
};

export default CVEPageDetails;
