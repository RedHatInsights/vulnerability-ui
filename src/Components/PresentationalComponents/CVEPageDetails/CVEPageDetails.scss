@import "../../../App";

.cve-title > div {
    max-width: 30%;
}

.cve-impact-badge {
    width: 1.8em;
}
