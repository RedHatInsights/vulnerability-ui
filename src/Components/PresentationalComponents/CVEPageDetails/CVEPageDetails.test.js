import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CVEPageDetails from './CVEPageDetails';

describe('CVEPageDetails component', () => {
    it('should render without data', () => {
        const data = {
            data: {},
            meta: {},
            isLoading: false
        };
        const wrapper = shallow(<CVEPageDetails data={ data } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with data', () => {
        const data = {
            data: {
                impact: 'Important',
                cvss3_score: '7.000',
                public_date: 'February 18th, 2019',
                description: 'Lorem ipsum',
                synopsis: 'CVE-2019-6454',
                package_list: [{
                    type: 'li',
                    key: 'systemd-resolved-219-62.el7_6.5.i686',
                    ref: null,
                    props: {
                        children: 'systemd-resolved-219-62.el7_6.5.i686'
                    },
                    _owner: null,
                    _store: {}
                }, {
                    type: 'li',
                    key: 'systemd-resolved-219-62.el7_6.5.x86_64',
                    ref: null,
                    props: {
                        children: 'systemd-resolved-219-62.el7_6.5.x86_64'
                    },
                    _owner: null,
                    _store: {}
                }, {
                    type: 'li',
                    key: 'systemd-devel-219-62.el7_6.5.ppc',
                    ref: null,
                    props: {
                        children: 'systemd-devel-219-62.el7_6.5.ppc'
                    },
                    _owner: null,
                    _store: {}
                }],
                rh_link: {
                    type: 'a',
                    key: null,
                    ref: null,
                    props: {
                        href: 'https://access.redhat.com/security/cve/CVE-2019-6454',
                        children: 'Red Hat CVE Database'
                    },
                    _owner: null,
                    _store: {}
                },
                mitre_link: {
                    type: 'a',
                    key: null,
                    ref: null,
                    props: {
                        href: 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-6454',
                        children: 'MITRE Database'
                    },
                    _owner: null,
                    _store: {}
                }
            },
            meta: {},
            isLoading: false
        };
        const wrapper = shallow(<CVEPageDetails data={ { data } } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with long description', () => {
        const data = {
            data: {
                impact: 'Important',
                cvss3_score: '7.000',
                public_date: 'February 18th, 2019',
                description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi consectetur molestie tristique.'
                    + ' Proin ornare urna ut consequat pellentesque. Sed commodo porta sollicitudin.'
                    + ' Interdum et malesuada fames ac ante ipsum primis in faucibus. Pellentesque habitant morbi'
                    + ' tristique senectus et netus et nullam.',
                synopsis: 'CVE-2019-6454',
                package_list: []
            },
            meta: {},
            isLoading: false
        };
        const wrapper = shallow(<CVEPageDetails data={ data } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with enabled WithLoader', () => {
        const data = {
            data: {},
            meta: {},
            isLoading: true
        };
        const wrapper = shallow(<CVEPageDetails data={ data } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
