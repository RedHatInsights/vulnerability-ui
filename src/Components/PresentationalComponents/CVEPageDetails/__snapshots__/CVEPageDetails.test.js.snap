// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVEPageDetails component should render with data 1`] = `
<Card
  className="cve-overview-card"
  component="article"
>
  <CardBody
    className=""
    component="div"
  >
    <Stack
      className=""
      component="div"
      gutter="md"
    >
      <StackItem
        className=""
        isMain={false}
      >
        <WithLoader>
          <CVEPageDetailsTitle
            publish_date=""
            title=""
          />
        </WithLoader>
      </StackItem>
      <StackItem
        className=""
        isMain={false}
      >
        <Grid
          className=""
          gutter={null}
          lg={null}
          md={null}
          sm={null}
          span={null}
          xl={null}
        >
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={8}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader>
              <CVEPageDetailsDescription
                description=""
                links={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              />
            </WithLoader>
          </GridItem>
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={1}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          />
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={3}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader>
              <CVEPageDetailsSeverity
                impact="Unknown"
              />
            </WithLoader>
          </GridItem>
        </Grid>
      </StackItem>
    </Stack>
  </CardBody>
</Card>
`;

exports[`CVEPageDetails component should render with enabled WithLoader 1`] = `
<Card
  className="cve-overview-card"
  component="article"
>
  <CardBody
    className=""
    component="div"
  >
    <Stack
      className=""
      component="div"
      gutter="md"
    >
      <StackItem
        className=""
        isMain={false}
      >
        <WithLoader
          loading={true}
        >
          <CVEPageDetailsTitle
            publish_date=""
            title=""
          />
        </WithLoader>
      </StackItem>
      <StackItem
        className=""
        isMain={false}
      >
        <Grid
          className=""
          gutter={null}
          lg={null}
          md={null}
          sm={null}
          span={null}
          xl={null}
        >
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={8}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader
              loading={true}
            >
              <CVEPageDetailsDescription
                description=""
                links={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              />
            </WithLoader>
          </GridItem>
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={1}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          />
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={3}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader
              loading={true}
            >
              <CVEPageDetailsSeverity
                impact="Unknown"
              />
            </WithLoader>
          </GridItem>
        </Grid>
      </StackItem>
    </Stack>
  </CardBody>
</Card>
`;

exports[`CVEPageDetails component should render with long description 1`] = `
<Card
  className="cve-overview-card"
  component="article"
>
  <CardBody
    className=""
    component="div"
  >
    <Stack
      className=""
      component="div"
      gutter="md"
    >
      <StackItem
        className=""
        isMain={false}
      >
        <WithLoader
          loading={false}
        >
          <CVEPageDetailsTitle
            publish_date="February 18th, 2019"
            title="CVE-2019-6454"
          />
        </WithLoader>
      </StackItem>
      <StackItem
        className=""
        isMain={false}
      >
        <Grid
          className=""
          gutter={null}
          lg={null}
          md={null}
          sm={null}
          span={null}
          xl={null}
        >
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={8}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader
              loading={false}
            >
              <CVEPageDetailsDescription
                description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi consectetur molestie tristique. Proin ornare urna ut consequat pellentesque. Sed commodo porta sollicitudin. Interdum et malesuada fames ac ante ipsum primis in faucibus. Pellentesque habitant morbi tristique senectus et netus et nullam."
                links={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              />
            </WithLoader>
          </GridItem>
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={1}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          />
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={3}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader
              loading={false}
            >
              <CVEPageDetailsSeverity
                cvss3_score="7.000"
                impact="Important"
              />
            </WithLoader>
          </GridItem>
        </Grid>
      </StackItem>
    </Stack>
  </CardBody>
</Card>
`;

exports[`CVEPageDetails component should render without data 1`] = `
<Card
  className="cve-overview-card"
  component="article"
>
  <CardBody
    className=""
    component="div"
  >
    <Stack
      className=""
      component="div"
      gutter="md"
    >
      <StackItem
        className=""
        isMain={false}
      >
        <WithLoader
          loading={false}
        >
          <CVEPageDetailsTitle
            publish_date=""
            title=""
          />
        </WithLoader>
      </StackItem>
      <StackItem
        className=""
        isMain={false}
      >
        <Grid
          className=""
          gutter={null}
          lg={null}
          md={null}
          sm={null}
          span={null}
          xl={null}
        >
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={8}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader
              loading={false}
            >
              <CVEPageDetailsDescription
                description=""
                links={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              />
            </WithLoader>
          </GridItem>
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={1}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          />
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={3}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <WithLoader
              loading={false}
            >
              <CVEPageDetailsSeverity
                impact="Unknown"
              />
            </WithLoader>
          </GridItem>
        </Grid>
      </StackItem>
    </Stack>
  </CardBody>
</Card>
`;
